.github/workflows/create-builds.yml

```diff
@@ -0,0 +1,116 @@
+# Summary:
+#
+# Reusable workflow for creating the distributable application assets.
+# Will also publish the application if the `app_type` input is `production`.
+#
+name: Create builds
+
+on:
+  workflow_call:
+    inputs:
+      app_type:
+        description: 'Type of application. Valid values are: internal, release-candidate, or production.'
+        required: true
+        type: string
+      build_sha:
+        description: 'SHA of commit to build'
+        required: true
+        type: string
+  workflow_dispatch:
+    inputs:
+      app_type:
+        description: 'Type of application. Valid values are: internal, or release-candidate'
+        required: true
+        type: choice
+        # We only allow internal and release candidate builds when doing manual dispatches.
+        # Production builds should only be triggered by the appropriate calling workflows.
+        options: ['internal', 'release-candidate']
+
+jobs:
+  validate_inputs:
+    name: Validate inputs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate app_type input
+        if: ${{ !(inputs.app_type == 'internal' || inputs.app_type == 'release-candidate' || inputs.app_type == 'production') }}
+        run: |
+          echo "app_type input must be one of the following: internal, release-candidate, production"
+          exit 1
+
+  builds:
+    name: Create builds
+    needs: validate_inputs
+    strategy:
+      matrix:
+        os:
+          [
+            { name: 'linux', image: 'ubuntu-latest' },
+            { name: 'macos', image: 'macos-latest' },
+            { name: 'macos-intel', image: 'macos-13' },
+            { name: 'windows', image: 'windows-latest' },
+          ]
+    runs-on: ${{ matrix.os.image }}
+    steps:
+      - name: Create GitHub App Token
+        uses: actions/create-github-app-token@v1
+        id: app-token
+        with:
+          app-id: ${{ vars.RELEASE_BOT_APP_ID }}
+          private-key: ${{ secrets.RELEASE_BOT_PRIVATE_KEY }}
+
+      - name: Checkout repository
+        env:
+          # When called from another workflow, the input value is used.
+          # When called from a manual dispatch, uses the value corresponding to the branch/tag selected in the GitHub UI.
+          BUILD_SHA: ${{ inputs.build_sha || github.sha }}
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.BUILD_SHA }}
+          fetch-depth: 0
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: 'npm'
+
+      - name: Create env file
+        env:
+          APP_TYPE: ${{ inputs.app_type }}
+        run: |
+          echo "APP_TYPE=${{ env.APP_TYPE }}" >> .env
+          echo "ONLINE_STYLE_URL=${{ secrets.ONLINE_STYLE_URL }}" >> .env
+
+          if [[ ${{ env.APP_TYPE }} != 'production' ]]; then
+            echo "VITE_FEATURE_TEST_DATA_UI=true" >> .env # We enable the test data creation UI for non-production builds
+          fi
```

Comment:
This seems very round-about? I thought the reason for using environment variables is to be able to set them in the environment.

Reply:
Thinking about dropping (3) as I look through what the main process is actually using those env variables for (starts [here](https://github.com/digidem/comapeo-desktop/blob/6d106db427b5dce6007ee5276e58693778b110fa/src/main/utils.js#L13-L19)). It probably doesn't make sense to have a file that defines those values at runtime now that I think about it.

Reply:
i think the `APP_TYPE` is also necessary for the main process to know at runtime, as it will be used for when we set up Sentry.

Reply:
I think for now, I'm going to keep this step. I can update the step name to be clearer about why we're creating an env file here.

Reply:
I think the issue with packaging a .env in the app to read configuration values is that a dev could easily put a sensitive value in there and not expect it to ship publicly. I think it's better to either bundle backend code and use a transform to replace selected env variables; add an app.config.json to source control and package with the app, or generate the app.config.json at build time based on env variables. I just think mixing the two is confusing as it's doing two separate things.

Reply:
I think the expo approach is pretty solid: a dynamic app.config.js which can read environment variables and writes the config to files that are not checked in, but are included at runtime (all-be-it in the expo case it's much more complicated that just a json - eg manifest.xml etc)

Reply:
> I think the issue with packaging a .env in the app to read configuration values is that a dev could easily put a sensitive value in there and not expect it to ship publicly.
> ...
>  I just think mixing the two is confusing as it's doing two separate things.

Yep both very true...

> I think it's better to either bundle backend code and use a transform to replace selected env variables

Yeah I think this is our best long-term option. Already have an issue for it here: #91 

> generate the app.config.json at build time based on env variables.

Think I'll opt for this approach for now.

Reply:
Removed this env file creation step via https://github.com/digidem/comapeo-desktop/pull/190/commits/fab883cf192488e2ba70e17005b6ac218a431bc8.

Planning to do the creation of the `app.config.json` that makes use of the env variables as a follow-up

Reply:
For posterity, stacked PR is here: https://github.com/digidem/comapeo-desktop/pull/191